app/code/Transoft/Blog/Controller/Adminhtml/Post/Save.php

```diff
@@ -0,0 +1,144 @@
+<?php
+declare(strict_types=1);
+
+namespace Transoft\Blog\Controller\Adminhtml\Post;
+
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\Action\HttpPostActionInterface;
+use Magento\Framework\App\ObjectManager;
+use Magento\Framework\App\Request\DataPersistorInterface;
+use Magento\Framework\Controller\ResultInterface;
+use Magento\Framework\Exception\CouldNotSaveException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Framework\View\Result\PageFactory;
+use Transoft\Blog\Api\ModelRepositoryInterface;
+use Transoft\Blog\Model\Model;
+use Transoft\Blog\Model\ModelFactory;
+use Transoft\Blog\Model\ModelRepository;
+
+/**
+ * Save post controller
+ */
+class Save extends Action implements HttpPostActionInterface
+{
+    const ADMIN_RESOURCE = 'Transoft_Blog::blog_manage_posts';
+
+    /**
+     * @var PageFactory $resultPageFactory
+     */
+    private $resultPageFactory;
+
+    /**
+     * @var ModelFactory
+     */
+    private $modelFactory;
+    /**
+     * @var DataPersistorInterface
+     */
+    private $dataPersistor;
+
+    /**
+     * @var ModelRepository
+     */
+    private $modelRepository;
+
+    /**
+     * Constructor
+     *
+     * @param Context $context
+     * @param PageFactory $resultPageFactory
+     * @param DataPersistorInterface $dataPersistor
+     * @param ModelFactory $modelFactory
+     * @param ModelRepositoryInterface $modelRepository
+     */
+    public function __construct(
+        Context $context,
+        PageFactory $resultPageFactory,
+        DataPersistorInterface $dataPersistor,
+        ModelFactory $modelFactory = null,
+        ModelRepositoryInterface $modelRepository = null
+    ) {
+        $this->resultPageFactory = $resultPageFactory;
+        $this->dataPersistor = $dataPersistor;
+        $this->modelFactory = $modelFactory
+            ?: ObjectManager::getInstance()->get(ModelFactory::class);
+        $this->modelRepository = $modelRepository
+            ?: ObjectManager::getInstance()->get(ModelRepositoryInterface::class);
+        parent::__construct($context);
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function execute()
+    {
+        $resultRedirect = $this->resultRedirectFactory->create();
+        $data = $this->getRequest()->getPostValue();
+
+        if ($data) {
+            if (empty($data['blog_id'])) {
+                $data['blog_id'] = null;
+            }
+
+            $post = $this->modelFactory->create();
+
+            $id = $this->getRequest()->getParam('id');
+            $image_url = $data['image_path'][0]['url'];
```

Comment:
What would be in a case when $data doesn't contain such keys

Reply:
added a guard here